### PR TITLE
[Go]:#2037 Completing structs concept exercise 

### PR DESCRIPTION
**Pull request by oanaOM**
Please review my additions to this exercise and let me know you thoughts.

### CODE REVIEW

File: languages/go/exercises/concept/structs/.docs/after.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+// create a new variable
+var dog Animal
+
+// assign a values to each field by using the . operator
+dog.name = "Rex"
+dog.age = 10
+
+// or you can assign the values to its fields individually
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/.docs/after.md:27]:
```suggestion
// or you can create a struct with field values
```

File: languages/go/exercises/concept/structs/.docs/after.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+dog := Animal{
+    name: "Bronson",
+    age: 7,
+}
+```
+
+Fields that don't have a initial value assigned, will have their zero value.
+
+To dive more deep into this type, you can check these resources: [Go by example: Structs], [A Tour of Go] or [Structures in Go (structs)]
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/.docs/after.md:36]:
```suggestion
To dive deeper into this type, you can check these resources: [Go by example: Structs], [A Tour of Go] or [Structures in Go (structs)]
```

File: languages/go/exercises/concept/structs/.docs/hints.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+## General
+
+A `structs` is a sequence of named elements, called fields, each one having a name and a type.
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/.docs/hints.md:3]:
```suggestion
A `struct` is a sequence of named elements, called fields, each one having a name and a type.
```

File: languages/go/exercises/concept/structs/structs.go [hunk 4]
```diff
@@ -1,20 +1,33 @@
...
+// Track implements a race track.
+type Track struct{
+	distance int
+}
+
+
+// NewTrack created a new track
+func NewTrack(distance int) Track{
+	panic("Please implement the NewTrack function") 
+}
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/structs.go:30]: My original plan was to let the student create this on their own. They have an example (`Car` and `NewCar`).

Do you think that leaves too much wiggle room for the student?

File: languages/go/exercises/concept/structs/.docs/hints.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+### 1. Creating a new remote controlled car
+
+- Define a new [struct][struct] with 3 fields
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/.docs/hints.md:7]:
```suggestion
- Instantiate a new [struct][struct] and fill the fields except `distance`.
```

File: languages/go/exercises/concept/structs/.docs/hints.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+- Create a function that returns the type of the new created struct and updates the fields accordingly
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/.docs/hints.md:8]:
```suggestion
- Create a function that returns the type of the newly created struct and update the fields accordingly
```

File: languages/go/exercises/concept/structs/.docs/hints.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+### 2. Creating a race track
+
+- Define a new [struct][struct] with 1 field
+- Create a function that returns the type of the new created struct and updates the fields accordingly
+
+### 3. Drive the car
+
+- Add a new field to keep track of the distance drive
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/.docs/hints.md:17]:
```suggestion
- Add a new field to keep track of the distance driven
```

File: languages/go/exercises/concept/structs/.docs/hints.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+- You need to check if there is enough battery to drive the car
+
+### 4. Check if a remote controlled car can finish a race
+
+- You need to calculate the maximum distance a car can drive
+- We assume that the car has initially the battery 100% charged
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/.docs/hints.md:23]:
```suggestion
- We assume that the cars' battery is charged to 100% initially
```

File: languages/go/exercises/concept/structs/.docs/introduction.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
+In Go, a `struct` is a sequence of named elements called _fields_, each field having a name and type. The name of a field must be unique within the struct. `Structs` can be compared with the _class_ in the Object Oriented Programming paradigm.
+
+You create a new struct by using the `struct` keyword, a **_built-in type_**, and explicty define the name and type of the field as shown un the example below.
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/.docs/introduction.md:3]:
```suggestion
You create a new struct by using the `struct` keyword, a **_built-in type_**, and explicty define the name and type of the fields as shown in the example below.
```

File: languages/go/exercises/concept/structs/.docs/introduction.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+
+```go
+type StructName struct{
+    field1 fieldType1
+    field2 fieldType2
+}
+```
+
+Struct fields are accessed using a `.` notation.
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/.docs/introduction.md:12]: maybe add a one-line example to this?

```go
someStruct.someField = "someValue"
```

File: languages/go/exercises/concept/structs/structs_test.go [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+	for _,tt :=range tests{
+		t.Run(tt.name, func(t *testing.T){
+			got := CanFinish(tt.car, tt.track)
+
+			if got!=tt.expected{
+				t.Errorf("NewCar(%v, %v) = %v; expected %v", tt.car,tt.track, got, tt.expected)
+			}
+		})
+	}
+}
```
Comment (tehsphinx) [inline languages/go/exercises/concept/structs/structs_test.go:187]: maybe add a newline after...

File: languages/go/exercises/concept/structs/structs.go (hunk 4 above)
Comment (oanaOM) [reply languages/go/exercises/concept/structs/structs.go:30]: No, it should be. Or we can have only the function declared and let them create the Track type.

**Comment by oanaOM** [comment]
@tehsphinx I'm very glad to hear this and thank you very much for your review. I've fixed the corrections and small changes if you could please review them as well so we can seal one more concept 😁  🚀

**Comment by tehsphinx** [comment]
Awesome work as always @oanaOM!

**Pull request merged**